examples/ArduinoIoTCloud-Schedule/ArduinoIoTCloud-Schedule.ino

```diff
@@ -0,0 +1,229 @@
+/*
+  This sketch demonstrates how to use the cloud schedule variable type.
+
+  This sketch is compatible with the following boards:
+   - MKR 1000
+   - MKR WIFI 1010
+   - MKR GSM 1400
+   - MKR NB 1500
+   - MKR WAN 1300/1310
+   - Nano 33 IoT
+   - ESP 8266
+*/
+
+#include "arduino_secrets.h"
+#include "thingProperties.h"
+
+#if defined(ESP32)
+static int const LED_BUILTIN = 2;
+#endif
+
+void setup() {
+  /* Initialize the serial port and wait up to 5 seconds for a connection */
+  Serial.begin(9600);
+  for(unsigned long const serialBeginTime = millis(); !Serial && (millis() - serialBeginTime > 5000); ) { }
+
+  /* Configure LED pin as an output */
+  pinMode(LED_BUILTIN, OUTPUT);
+
+  /* This function takes care of connecting your sketch variables to the ArduinoIoTCloud object */
+  initProperties();
+
+  /* Initialize Arduino IoT Cloud library */
+  ArduinoCloud.begin(ArduinoIoTPreferredConnection);
+
+  setDebugMessageLevel(DBG_INFO);
+  ArduinoCloud.printDebugInfo();
+
+  /* Setup one shot schedule example */
+  setupOneShotSchedule();
+
+  /* Setup per minute schedule example */
+  setupMinuteSchedule();
+
+  /* Setup hourly schedule example */
+  setupHourlySchedule();
+
+  /* Setup daily schedule example */
+  setupDailySchedule();
+
+  /* Setup weekly schedule example */
+  setupWeeklySchedule();
+
+  /* Setup monthly schedule example */
+  setupMonthlySchedule();
+
+  /* Setup yearly schedule example */
+  setupYearlySchedule();
+}
+
+ /* Setup a schedule with an active period of 5 minutes that doesn't repeat
+  * Starting from 2021 11 01 17:00:00
+  * Until         2021 11 02 17:00:00
+  */
+void setupOneShotSchedule() {
+
+  unsigned int startingFrom = TimeService::getTimeFromString("2021 Nov 01 17:00:00");
+  unsigned int until = startingFrom + ( DAYS * 1 );
+  unsigned int activePeriod = MINUTES * 5;
+
+  /* Warning: there is no cross check between until and activePeriod */
+  unsigned int scheduleConfiguration =  Schedule::createOneShotScheduleConfiguration();
```

Comment:
I was thinking: Why not create a `typedef` to hide the fact that a schedule is nothing more than an `unsigned int`? If at some point in time it becomes a class there's no need change anything, as it already appears like a class due to the `typedef`, i.e.
```C++
typedef unsigned int CloudScheduleType;
```
This could be combined with a `static_assert` to ensure that the cloud type always has the minimum required value, i.e.
```C++
static_assert(sizeof(CloudScheduleType int) >= 4", Error, sizeof(CloudScheduleType) < 4, can't use CloudSchedule on this platform."), 
```

Reply:
I'm not sure i get the point, `Schedule` and `CloudSchedule` are already two classes; in the example `scheduleConfiguration` is just one parameter needed to initialize the schedule. That said i think using a typedef for the scheduleConfiguration and the other input parameters is really a good idea so i've created a commit you can check. https://github.com/pennam/ArduinoIoTCloud/commit/2b63f01f2e41c2ad9d75729087eb95bbbf0960e9

Reply:
No, you've gotten me perfectly. :+1: for `ScheduleTimeType` and `ScheduleConfigurationType`.

Reply:
-> see https://github.com/arduino-libraries/ArduinoIoTCloud/pull/284/commits/210edd275792da456221fb8d413a884cbcb624be